client/landing/stepper/declarative-flow/new-hosted-site-flow.ts

```diff
@@ -87,6 +88,10 @@ const hosting: Flow = {
 					setSignupCompleteSlug( providedDependencies?.siteSlug );
 					setSignupCompleteFlowName( flowName );
 
+					if ( ! providedDependencies.goToCheckout ) {
+						recordFreeHostingTrialStarted( flowName );
+					}
```

Comment:
This looks fine to be because I have worked on that part of Calypso, but I wonder if we could be more explicit here. Instead of inferring whether the site is a trial because it doesn't go to checkout, what if we compared the provided plan slug? It'd be clearer.

Reply:
Thanks! I've added a more explicit check for whether it's a free trial, and not relying on everything not ending up at the checkout page being a free trial. I've never worked on these flows before, so I'm not sure if this logic is the right one, but I tested a few times going through the flow with either a trial or a paid plan, and it seems it's working.

There may be better ways of fetching the plan slug?

Reply:
I think the way you're doing it is fine.

Reply:
I would use a full if instead of the short-hand. It's such a foot gun to do it like `condition && command` outside of React components.